app/Http/Controllers/HomeController.php

```diff
@@ -55,7 +56,40 @@ public function downloadQuotaCheck()
 
     public function getDownload()
     {
-        return ext_view('home.download');
+        static $lazerPlatformNames = [
+            'android' => 'Android 5 or later',
+            'ios' => 'iOS 10 or later',
+            'linux_x64' => 'Linux (x64)',
+            'macos_as' => 'macOS 10.15 or later (Apple Silicon)',
+            'windows_x64' => 'Windows 8.1 or later (x64)',
+        ];
+
+        $httpHeaders = [];
+        // format headers to what Agent is expecting
+        foreach (request()->headers->all() as $key => $values) {
+            $headerKey = 'HTTP_'.strtoupper(strtr($key, '-', '_'));
+            $httpHeaders[$headerKey] = $values[0];
+        }
+        $agent = new Agent($httpHeaders);
+
+        if ($agent->is('Windows')) {
+            $platform = 'windows_x64';
+        } elseif ($agent->isiOS() || $agent->isiPadOS()) {
+            $platform = 'ios';
+        } elseif ($agent->is('OS X')) {
+            // FIXME: figure out a way to differentiate Intel and Apple Silicon
+            $platform = 'macos_as';
+        } elseif ($agent->is('Linux')) {
+            $platform = 'linux_x64';
+        } elseif ($agent->isAndroidOS()) {
+            $platform = 'android';
+        }
+        $platform ??= 'windows_x64';
```

Comment:
that's an interesting way of writing `else { $platform = 'windows_x64' }` (or use `match` since every case is 1 line 🤔 )

Reply:
it was originally a bunch of nested ifs but then it's not anymore